docs/guides/how_to_guides/creating_batches/how_to_load_a_spark_dataframe_as_a_batch.rst

```diff
@@ -85,105 +85,176 @@ This guide will help you load a Spark DataFrame as a Batch for use in creating E
 
-        3. Construct a BatchRequest
+          Example ``great_expectations.yml`` Datsource configuration:
 
-            We will create a ``BatchRequest`` and pass it our DataFrame via the ``batch_data`` argument.
+          .. code-block:: yaml
 
-            Attributes inside the ``data_connector_query`` are optional - you can use them to attach additional metadata to your DataFrame. When configuring the Data Connector, you used ``batch_identifiers`` to define which keys are allowed.
+            my_spark_datasource:
+              execution_engine:
+                module_name: great_expectations.execution_engine
+                class_name: SparkExecutionEngine
+              module_name: great_expectations.datasource
+              class_name: Datasource
+              data_connectors:
+                my_runtime_data_connector:
+                  class_name: RuntimeDataConnector
+                  batch_identifiers:
+                    - some_key_maybe_pipeline_stage
+                    - some_other_key_maybe_airflow_run_id
 
-            .. code-block:: python
+          Example Runtime Batch Request using an in-memory DataFrame:
 
-                from great_expectations.core.batch import BatchRequest
-
-                batch_request = BatchRequest(
-                    datasource_name="insert_your_spark_datasource_name_here",
-                    data_connector_name="insert_your_runtime_data_connector_name_here",
-                    data_asset_name="insert_your_data_asset_name_here",
-                    batch_data=insert_your_dataframe_here,
-                    data_connector_query={
-                        "batch_identifiers": {
-                            "some_key_maybe_pipeline_stage": "ingestion step 1",
-                            "some_other_key_maybe_run_id": "run 18"
-                        }
+          .. code-block:: python
+
+            from great_expectations.core.batch import RuntimeBatchRequest
+            from great_expectations.core.util import get_or_create_spark_application
+            import pyspark
+
+            spark_application: pyspark.sql.session.SparkSession = get_or_create_spark_application()
+            df: pyspark.sql.dataframe.DataFrame = spark_application.read.csv("some_path.csv")
+            runtime_batch_request = RuntimeBatchRequest(
+                datasource_name="my_spark_datasource",
+                data_connector_name="my_runtime_data_connector",
+                data_asset_name="insert_your_data_asset_name_here",
+                runtime_parameters={
+                  "batch_data": df
+                },
+                batch_identifiers={
+                    "some_key_maybe_pipeline_stage": "ingestion step 1",
+                    "some_other_key_maybe_run_id": "run 18"
+                }
```

Comment:
similar story with the mis-matched keys here

Reply:
Fixed